src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -111,11 +129,43 @@ export class TextInputComponent implements OnInit {
     return values.map(d => (d < 10 ? '0' + d : d)).map(String)
   }
 
+  datePickerObj: any = {}
+  selectedDate: string = new Date().toLocaleDateString('en-ZA')
+
+  async openDatePicker() {
+    const datePickerModal = await this.modalCtrl.create({
+      component: Ionic4DatepickerModalComponent,
+      cssClass: 'li-ionic4-datePicker',
+      componentProps: {
+        objConfig: this.datePickerObj,
+        selectedDate: this.selectedDate
+      }
+    })
+    await datePickerModal.present()
+
+    datePickerModal.onDidDismiss().then(data => {
+      console.log(data)
+      this.selectedDate = data.data.date
+
+      // set defaultDatepickervalue
+      const split_date = data.data.date.split('-')
```

Comment:
What is the format of the date returned?

Reply:
I set it to "YYYY-MM-DD" so I can parse the data with `split`

Reply:
alternative: 
```
const date = moment(data.data.date)
this.defaultDatePickerValue = {
  year: date.format('YYYY'),
  month: date.format('M'),
  day: date.format('D'),
}
```

Reply:
Yes I think this is a good alternative solution.